### PR TITLE
chore: Adding a check for old version of the CLI

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: packages/angular-cli/upgrade/version.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import {CliConfig} from '../models/config';
+import {readFileSync, existsSync} from 'fs';
+import {stripIndents} from 'common-tags';
+import {bold, red, yellow} from 'chalk';
+import * as path from 'path';
+
+
+function _findUp(name: string, from: string) {
```
Comment (Brocco) [inline packages/angular-cli/upgrade/version.ts]: Why is this not a private static method inside the Version class?

File: packages/angular-cli/upgrade/version.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    currentDir = path.dirname(currentDir);
+  }
+
+  return null;
+}
+
+
+function _hasOldCliBuildFile() {
```
Comment (Brocco) [inline packages/angular-cli/upgrade/version.ts]: Why is this not a private static method inside the Version class?

Comment (hansl) [inline packages/angular-cli/upgrade/version.ts]: It doesn't access any class property/method, so it doesn't need to be a version.

File: packages/angular-cli/upgrade/version.ts (hunk 1 above)
Comment (hansl) [inline packages/angular-cli/upgrade/version.ts]: See below.

File: packages/angular-cli/commands/build.ts [hunk 3]
```diff
@@ -42,6 +43,8 @@ const BuildCommand = Command.extend({
       }
     }
 
+    Version.assertPostWebpackVersion();
```
Comment (filipesilva) [inline packages/angular-cli/commands/build.ts]: This should not be only on builds, but on calls to `ng`. Testing and generators also have changed and assume post-webpack.

File: packages/angular-cli/models/config.ts [hunk 4]
```diff
@@ -39,15 +39,15 @@ export class CliConfig extends CliConfigBase<ConfigInterface> {
 
 
   static fromProject(): CliConfig {
```
Comment (filipesilva) [inline packages/angular-cli/models/config.ts]: This gives you the version the project was generated with, not the version it is currently using. You can only know that from the installed package.

File: packages/angular-cli/commands/build.ts (hunk 3 above)
Comment (hansl) [inline packages/angular-cli/commands/build.ts]: Yes, sorry I forgot that. Let me fix.

File: packages/angular-cli/models/config.ts (hunk 4 above)
Comment (hansl) [inline packages/angular-cli/models/config.ts]: Good point, I'll fix.

**Pull request merged**